File: CS/WindowsFormsApp1/Employees.cs
using DevExpress.DataAccess.ObjectBinding;
using System.Collections.Generic;
using System.ComponentModel;
// ...
public class Employee
{
    public string Name { get; set; }
    public string Position { get; set; }
}

[DisplayName("Employees")]
[HighlightedClass]
public class EmployeeDataSource
{
    private string department;
    private List<Employee> management = new List<Employee>() {
        new Employee() {
            Name = "Ana Trujillo",
            Position = "CEO"
        },
        new Employee() {
            Name = "Andrew Fuller",
            Position = "Vice President, Sales"
        }
    };
    private List<Employee> financial = new List<Employee>() {
        new Employee() {
            Name = "Nancy Davolio",
            Position = "Accountant"
        },
        new Employee() {
            Name = "Maria Anders",
            Position = "Accountant"
        }
    };
    private List<Employee> sales = new List<Employee>() {
        new Employee() {
            Name = "Antonio Moreno",
            Position = "Sales Representative"
        },
        new Employee() {
            Name = "Thomas Hardy",
            Position = "Sales Representative"
        },
        new Employee() {
            Name = "Christina Berglund",
            Position = "Order Administrator"
        },
        new Employee() {
            Name = "Frédérique Citeaux",
            Position = "Marketing Manager"
        },
        new Employee() {
            Name = "Hanna Moos",
            Position = "Sales Representative"
        }
    };
    public EmployeeDataSource()
    {
        this.department = "Management";
    }

    [HighlightedMember]
    public EmployeeDataSource(string department)
    {
        this.department = department;
    }

    [HighlightedMember]
    public IEnumerable<Employee> GetEmployeeList(int recordCount)
    {
        List<Employee> employees = new List<Employee>();
        if (this.department == "Management")
            employees = this.management;
        if (this.department == "Financial")
            employees = this.financial;
        if (this.department == "Sales")
            employees = this.sales;
        foreach (var employee in employees)
            if (recordCount-- > 0)
                yield return employee;
    }
}

File: VB/WindowsFormsApp1/Employees.vb
Imports DevExpress.DataAccess.ObjectBinding
Imports System.Collections.Generic
Imports System.ComponentModel
' ...
Public Class Employee
	Public Property EmployeeName() As String
	Public Property EmployeePosition() As String
End Class

<DisplayName("Employees"), HighlightedClass>
Public Class EmployeeDataSource
	Private department As String
	Private management As New List(Of Employee)() From {
		New Employee() With {
			.EmployeeName = "Ana Trujillo",
			.EmployeePosition = "CEO"
		},
		New Employee() With {
			.EmployeeName = "Andrew Fuller",
			.EmployeePosition = "Vice President, Sales"
		}
	}
	Private financial As New List(Of Employee)() From {
		New Employee() With {
			.EmployeeName = "Nancy Davolio",
			.EmployeePosition = "Accountant"
		},
		New Employee() With {
			.EmployeeName = "Maria Anders",
			.EmployeePosition = "Accountant"
		}
	}
	Private sales As New List(Of Employee)() From {
		New Employee() With {
			.EmployeeName = "Antonio Moreno",
			.EmployeePosition = "Sales Representative"
		},
		New Employee() With {
			.EmployeeName = "Thomas Hardy",
			.EmployeePosition = "Sales Representative"
		},
		New Employee() With {
			.EmployeeName = "Christina Berglund",
			.EmployeePosition = "Order Administrator"
		},
		New Employee() With {
			.EmployeeName = "Frédérique Citeaux",
			.EmployeePosition = "Marketing Manager"
		},
		New Employee() With {
			.EmployeeName = "Hanna Moos",
			.EmployeePosition = "Sales Representative"
		}
	}
	Public Sub New()
		Me.department = "Management"
	End Sub

	<HighlightedMember>
	Public Sub New(ByVal department As String)
		Me.department = department
	End Sub

	<HighlightedMember>
	Public Iterator Function GetEmployeeList(ByVal recordCount As Integer) As IEnumerable(Of Employee)
		Dim employees As New List(Of Employee)()
		If Me.department = "Management" Then
			employees = Me.management
		End If
		If Me.department = "Financial" Then
			employees = Me.financial
		End If
		If Me.department = "Sales" Then
			employees = Me.sales
		End If
		For Each employee In employees
			Dim tempVar As Boolean = recordCount > 0
			recordCount -= 1
			If tempVar Then
				Yield employee
			End If
		Next employee
	End Function
End Class
